src/app/clusters/commissioner-control-server/commissioner-control-server.h

```diff
@@ -0,0 +1,180 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace CommissionerControl {
+
+// Set to 64 bytes, which is more than enough for storing an IPv6 address in either binary or text form.
+static constexpr size_t kIpAddressBufferSize = 64;
+
+struct CommissioningApprovalRequest
+{
+    uint64_t requestId;
+    VendorId vendorId;
+    uint16_t productId;
+    NodeId clientNodeId;
+    FabricIndex fabricIndex;
+    Optional<CharSpan> label;
+};
+
+struct CommissioningWindowParams
+{
+    uint32_t iterations;
+    uint16_t commissioningTimeout;
+    uint16_t discriminator;
+    ByteSpan PAKEPasscodeVerifier;
+    ByteSpan salt;
+};
+
+struct CommissionNodeInfo
+{
+    CommissioningWindowParams params;
+    Optional<ByteSpan> ipAddress;
```

Comment:
Strongly suggest making ipAddress protected and having a getter for a const thing, to avoid people accidentally creating security bugs.